cmd/registry/controller/controller_test.go

```diff
@@ -774,6 +778,30 @@ func TestMultipleEntitiesArtifacts(t *testing.T) {
 			}
 			actions := ProcessManifest(ctx, lister, projectID, manifest, 10)
 
+			if test.generatedResourceRequiresRevision {
+				for i := range test.want {
+					gr := test.want[i].GeneratedResource
+					a, err := names.ParseArtifact(gr)
+					if err != nil {
+						t.Fatal("Failed to parse GeneratedResource", err)
+					}
+					sr := names.SpecRevision{
+						ProjectID:  a.ProjectID(),
+						ApiID:      a.ApiID(),
+						VersionID:  a.VersionID(),
+						SpecID:     a.SpecID(),
+						RevisionID: a.RevisionID(),
```

Comment:
This flow is a little confusing. We parse the `GeneratedResource` (which doesn't have a revisionID) into a `SpecRevision`, fetch the `SpecRevision` and use that revisionID to populate the `GeneratedResource`.

Specially the `RevisionID: a.RevisionID()` is quite confusing. Can we instead parse the `GeneratedResource` into a `Spec` -> fetch that spec and then use the spec's revision ID to populate the `GeneratedResource`. This type of flow will avoid the confusion I think.

Reply:
Sounds good.